ui/src/app/pages/access/data-access-requirements.spec.tsx

```diff
@@ -2083,4 +2088,26 @@ describe('DataAccessRequirements', () => {
 
     expectButtonElementEnabled(screen.queryByText('Confirm'));
   });
+
+  it('should show an error modal when syncModulesExternal fails', async () => {
+    const spySyncModulesExternal = jest.spyOn(
+      accessUtils,
+      'syncModulesExternal'
+    );
+    spySyncModulesExternal.mockImplementation(() =>
+      Promise.reject('Deliberate testing failure for syncModulesExternal')
+    );
+    component();
+    expect(spySyncModulesExternal).toHaveBeenCalledTimes(1);
+    expect(notificationStore.get()).toBeNull();
```

Comment:
I don't know how I feel about the use about using notificationStore in tests, because it is not testing visual behavior. However this component does not render the modal. So we need to rely on the notificationStore (an implementation detail) to test this behavior. The same can be said for the spinner, because this component doesn't render the spinner.

I am up for

Reply:
Agreed. I think checking the store directly is fine in this PR, but here's a couple alternatives we could consider:
- Add a notification-aware component rendering test helper. See the 45:00 mark of [this recording](https://drive.google.com/file/d/111ZvJV4wu1dvapW30LUa5cJZZhkz05_T/view) for inspiration.
- Re-structure notifications so pages that show notifications _do_ render the notification component.

Reply:
Minor: As is, it seems like there's a race condition at a glance. I'd prefer a separate test for the success case that asserts it stays null or performing the null check before creating the component.